src/layout/src/components/App/AppHeaderWrapper.js

```diff
@@ -363,53 +356,186 @@ class AppHeaderWrapper extends Component<Props, State> {
                 value={selectedOrganizationOption} />
           )
         }
-        <LogoutButton onClick={logout}>Log Out</LogoutButton>
-      </HeaderRightContentWrapper>
+        {
+          renderLogOutButton
+            ? (
+              <LogoutButton onClick={logout}>Log Out</LogoutButton>
+            )
+            : (
+              <NavigationToggleWrapper onClick={this.toggleNavigation}>
+                <FontAwesomeIcon icon={faBars} />
+              </NavigationToggleWrapper>
+            )
+        }
+      </HeaderSectionContentWrapper>
     );
   }
 
   render() {
 
-    const { children, className, icon } = this.props;
-    const { shouldWrapNav } = this.state;
+    const {
+      appIcon,
+      appTitle,
+      children,
+      className,
+    } = this.props;
+    const { isNavigationOpen, shouldWrapNavigation } = this.state;
+    const { handleNavigationWrapping, navigationChildrenCount } = this.processChildren();
 
-    const childrenCount = Children.count(children);
+    let headerBounds;
+    if (this.headerRef.current) {
+      headerBounds = this.headerRef.current.getBoundingClientRect();
+    }
 
     return (
       <>
-        <AppHeaderOuterWrapper className={className} ref={this.headerRef}>
+        <AppHeaderOuterWrapper className={className} ref={this.headerRef} onClick={this.handleOnClickHeader}>
           <AppHeaderInnerWrapper>
             {
-              childrenCount > 0 && (
-                <NavigationContentWrapper ref={this.nav1Ref}>
-                  {
-                    Children.map(children, (child, index) => {
-                      if (index === 0) {
-                        return React.cloneElement(
-                          child,
-                          { ...child.props, className: NAV_ROOT },
-                          React.createElement(AppIcon, { icon }),
-                          React.createElement(AppTitle, { title: child.props.children }),
-                        );
-                      }
-                      return shouldWrapNav ? null : child;
-                    })
-                  }
-                </NavigationContentWrapper>
+              /*
+               * this block is expected to handle the following examples:
+               *
+               *   1. this is unlikely to be common and is not ideal
+               *     <AppHeaderWrapper ... />
+               *
+               *   2. this is unlikely to be common and should be avoided
```

Comment:
These are interesting things to avoid. If it's intended for the AppHeaderWrapper to always have an AppNavigationWrapper as a child, why are they separate? All of the following blocks specifically expect this pattern.

Reply:
Well... I'm exposing `AppNavigationWrapper` to users of LUK, so the following is also possible (and should just work):

```
<AppHeaderWrapper ...>...</AppHeaderWrapper>
<AppNavigationWrapper ...>...</AppNavigationWrapper>
<AppNavigationWrapper ...>...</AppNavigationWrapper>
```

It's preferred for `AppNavigationWrapper` to be a child because the header can do the wrapping magic, but everything should function just fine if it's not a child.